### DISCUSSION TITLE
Support Deployless Reads

### DISCUSSION THREAD

**Discussion by hammeiam** [wevm/wagmi#4160] [Ideas] [answered]
[Deployless Reads are supports by Viem](https://viem.sh/docs/contract/readContract#deployless-reads), and it would be nice to have Wagmi support for the same functionality via ReadContract rather than extracting the underlying viem client.

**Comment by tmm** [comment, marked as answer]
It's already supported, just hasn't been carried over to the Wagmi docs yet. (PR welcome for that!)

[TypeScript Playground](https://www.typescriptlang.org/play/?target=99#code/JYWwDg9gTgLgBAbzgUwHYGMIBNkDECuGMwEqAIgIYwUA0cYFUAzsgIIBGwcAvnAGZQIIOAHIAbsGQgRAKFCRYiOPhYAlZBSwBhUjCgV08XgKGiA7hQDmIYLJkB6e3ABCATxjJMOGZlRN4UMhM+AA28AC8ymoa2rr6hgAUCDJwcBScAFz0jCwcwAkA2iJ8hIYkqHCoFCDICQCUcBLIZnCBMPhQfnAJ-lDAqJZ1IgC6dTQpcF7IWSIADAAeAHTLInSOcKzo6EFMwOwhyHAAFsiBWUcwMGBMGY7IMCfM6BSoiyT2mliBTEz2C7jOVgAZgAnABGABMWgoswALFgsLCAKwADnQKNhQOQEIAojjZiiQQA2ADsfC0cL4QKRJJB6AhAGIphMSkRygA5arTURVGqrGTcOoyb6hGCLLBUCgyBxOMjIMAhCCuOC4AwwaCuHykfxpLYQQgRRATT7fG6iBYk2YgkG0y3oJFglGQrDoIEwsGzdAkrCzMHsJGzZCzLBg71euko-mpdLALIMZhsTiFYqlYikFCoPSuAAKEH6MHqjUkLTaHS6CRNOyGo3GvAoTEm2pgWr8ASCoohcEiKmQ6k0Okz8QLyVS61YCNNcAgfDgD0OAGUQIx4JtMAa4DgFUrTt0+GqNXVFiz91BXDMFjiUWQ+CiIXxWLCcR7kETWJCKCDKP6kRD2NtWBCBhYBCRJyiC7DsPyEzrAQbLpuqKDzJ4+AeFOFRzvwJ7Kghm6KthJxwIuy4bHqBpHqke6GBqlDUFkaBTLBZSkDRFBJBM0aZNkCZ5MmrJMRU6CBFQbCkZmFYTjsU5mKgpx0Pg+YQkiRJwEwFBhNWYzsfwqYclyMyCRoHirvqmZRqkaRQJYZoFAYa6ZoslY-HQsyoMM4ypIK4zQU4jFpuhECTGpIRobOBFEYoxlkcanG2SZYoxu5aQST8WSxWRjlMIlfF+ZyNQzGgWa5vm-KCsK7ZhBC4qSjIQA)

```ts
import { encodeFunctionData, parseAbi } from 'viem'
import { useReadContract } from 'wagmi'

// Bytecode
const result = useReadContract({
  abi: parseAbi(['function name() view returns (string)']),
  code: '0x...', // Accessible here: https://etherscan.io/address/0xFBA3912Ca04dd458c843e2EE08967fC04f3579c2#code
  functionName: 'name',
})

// Deploy Factory
const account = {
  address: '0x70997970c51812dc3a010c7d01b50e0d17dc79c8',
  abi: parseAbi(['function entryPoint() view returns (address)']),
} as const
const result2 = useReadContract({
  // Address of the Smart Account deployer (factory).
  factory: '0xE8Df82fA4E10e6A12a9Dab552bceA2acd26De9bb',

  // Function to execute on the factory to deploy the Smart Account.
  factoryData: encodeFunctionData({
    abi: parseAbi(['function createAccount(address owner, uint256 salt)']),
    functionName: 'createAccount',
    args: [account.address, 0n],
  }),

  // Function to call on the Smart Account.
  abi: account.abi,
  address: account.address,
  functionName: 'entryPoint',
})
```

**Reply by hammeiam** [reply]
I'm still having some issues... 
First, `code` is not recognized by typescript, which I guess is to be expected if it hasn't made it into the docs yet?
Second, it looks like not have setting an `address` would force the call to be disabled [[code]](https://github.com/wevm/wagmi/blob/main/packages/react/src/hooks/useReadContract.ts#L87) so I'm not sure how your demo is running

**Reply by tmm** [reply]
Make sure you are using the latest version of Viem. If this doesn't work, feel free to open an issue with a minimal reproduction.
